File: Cpp/118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> arr;//(numRows, vector<int>());
        for(int i = 0; i < numRows; i++){
            arr.push_back(vector<int>(i+1, 0));
            for(int j = 0; j <= i; j++){
                cout<<j<<endl;
                if(j == 0 || i == j){
                    arr[i][j] = 1;
                }
                else{
                    arr[i][j] = (arr[i-1][j]+arr[i-1][j-1]);
                }                
            }
        }
        return arr;
    }
};

File: Python/118.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        sol = [[1]]
        i = 2
        while i <= numRows:
            sol.append([])
            for j in range(i):
                if j == 0:
                    sol[-1].append(1)
                elif j == i-1:
                    sol[-1].append(1)
                else:
                    sol[-1].append(sol[-2][j]+sol[-2][j-1])
            i+=1
        return sol
